vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueKind
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueKind Kind;
+    UInt64 Data;
+  };
+
+  struct JsiSymbolData
```

Comment:
>JsiSymbolData [](start = 9, length = 13)

@Dannyvv: Figure out difference between Symbol and PropertyName

Reply:
@dannyvv, my understanding is that the PropNameID is the Hermes engine specific optimization. It seems that it is an internal optimization exposed to the public API. All non-Hermes implementations use JavaScript Strings for PropNameID. The Symbol is another story - they allow to define 'private' properties which are not accessible unless you have exactly the same Symbol instance (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol). I am considering to remove the PropNameID from the ABI API and map the JSI PropNameID to the JS String instead.